nb2/bot/slack_bot.py

```diff
@@ -118,6 +127,28 @@ def remember(self, target_slack_user_id: str, quote: str):
     # Action matching functions
     #############################
 
+    def is_hello(self, command: str) -> bool:
+        """
+        Return True if the command string is a greeting.
+
+        The valid form for a hello is:
+        "<@NB_user_id> <greeting>"
+
+        where <greeting> is a word in valid_greetings.
+        """
+        valid_greetings = ["hello", "greetings", "salutations", "howdy"]
+        content = [token for token in command.split()]
+
+        if len(content) > 2:
+            return False
+
+        cleaned_content = [re.sub(r"\W+", "", token).lower() for token in content]
+
+        if cleaned_content[1] in valid_greetings:
```

Comment:
I still think we should allow for a response if the command doesn't start with <@NB_user_id> so that the bot can be DMed, but maybe that's a future refactor.

Reply:
(doing for later as suggested)